src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -407,18 +407,17 @@ private void NotifyLeaderAboutRemoval(string dbName, string databaseId)
                         var ex = t.Exception.ExtractSingleInnerException();
                         if (ex is DatabaseDoesNotExistException)
                             return;
-
-                        if (_logger.IsInfoEnabled)
-                        {
-                            _logger.Info(message, t.Exception);
-                        }
                     }
 
-                    if (t.IsCanceled)
+                    try
+                    {
+                        await t; // throw immediately
```

Comment:
it is tricky now, IMO it should be inside the t.IsFaulted check?

Reply:
it might be canceled

Reply:
what is the reason of cancellation? because you are spawning a command few lines below... it will also get canceled?

Reply:
Also the `await Task.Delay(TimeSpan.FromSeconds(15));` should have some sort of a server shutdown passed to avoid waiting 15 seconds in case of database shutdown (with ex handling) ?